### PR TITLE
Add bare minimum documentation

### PR DESCRIPTION
**Pull request by vamsiampolu**
@MicheleBertoli I took your comments from the issue and added it to the README https://github.com/styled-components/jest-styled-components/issues/33

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,10 +5,64 @@
 
 ## Installation
 
+
 ```
 yarn add --dev jest-styled-components
 ```
 
+This package must be used with either `react-test-renderer` or `enzyme` and `enzyme-to-json` to generate a serialized `tree` from your component.
```
Comment (MicheleBertoli) [inline README.md]: This should be more generic, like "To render React components for testing you can use either [...]" because it doesn't apply to this package only.

Also, `enzyme-to-json` is only needed for snapshots testing.

File: README.md [hunk 2]
```diff
@@ -5,10 +5,64 @@
...
+yarn add --dev react-test-renderer
+```
+
+Usage:
+
+```js
+import renderer from 'react-test-renderer'
+
+// somewhere in your tests
+it('matches the styled components snapshot', () => {
```
Comment (MicheleBertoli) [inline README.md]: nit: `test(`

File: README.md [hunk 3]
```diff
@@ -5,10 +5,64 @@
...
+  const tree = renderer.create(<MyComponent />).toJSON()
+  expect(tree).toMatchStyledComponentsSnapshot()
```
Comment (MicheleBertoli) [inline README.md]: nit: new line

File: README.md [hunk 4]
```diff
@@ -5,10 +5,64 @@
...
+  expect(tree).toHaveStyleRule('display', 'block')
+})
+```
+
+For more info on using react-test-renderer with jest for snapshots see [this](https://facebook.github.io/jest/docs/snapshot-testing.html)
```
Comment (MicheleBertoli) [inline README.md]: Can we rephrase this to "Lear more about [Snapshot Testing](https://facebook.github.io/jest/docs/snapshot-testing.html) with Jest." or something similar?

Also, I [personally wrote](https://github.com/facebook/jest/pull/3425) the FAQs in that page :)

File: README.md [hunk 5]
```diff
@@ -5,10 +5,64 @@
...
+
+Installation:
+
+```
+yarn add --dev enzyme enzyme-to-json
+```
+
+Usage:
+
+This module is not compatible with shallow rendering, components must use the full DOM rendering with `mount`
```
Comment (MicheleBertoli) [inline README.md]: This is not correct, the module works with both [shallow](https://github.com/styled-components/jest-styled-components/blob/master/test/index.spec.js#L46) and [mount](https://github.com/styled-components/jest-styled-components/blob/master/test/index.spec.js#L56).

File: README.md [hunk 6]
```diff
@@ -5,10 +5,64 @@
...
+
+```js
+import {mount} from 'enzyme'
+import toJSON from 'enzyme-to-json'
+
+// inside your test
+it('can use enzyme instead of react-test-renderer', () => {
```
Comment (MicheleBertoli) [inline README.md]: nit: `test(`

File: README.md [hunk 7]
```diff
@@ -5,10 +5,64 @@
...
+yarn add --dev enzyme enzyme-to-json
+```
+
+Usage:
+
+This module is not compatible with shallow rendering, components must use the full DOM rendering with `mount`
+
+```js
+import {mount} from 'enzyme'
+import toJSON from 'enzyme-to-json'
```
Comment (MicheleBertoli) [inline README.md]: `enzyme-to-json` can be used like this in each test file, or it can be [added globally](https://github.com/styled-components/jest-styled-components/blob/master/package.json#L54) (which is more convenient).

File: README.md [hunk 8]
```diff
@@ -29,6 +83,9 @@ expect(tree).toMatchStyledComponentsSnapshot()
 
 ## toHaveStyleRule [React]
 
+Only checks for the styles directly applied to the component it receives, to assert that a style has been applied to a child
```
Comment (MicheleBertoli) [inline README.md]: > to assert that a style has been applied to a child component

You can make assertion on a child component using Enzyme's [find](https://github.com/airbnb/enzyme/blob/master/docs/api/ReactWrapper/find.md).
Snapshots should be used for testing [complex selectors](https://github.com/styled-components/jest-styled-components/issues/33#issue-237062310) and/or the entire output in one go.

**Comment by wongterrencew** [comment]
Thanks @vamsiampolu this has helped me too!

File: README.md (hunk 5 above)
Comment (vamsiampolu) [inline README.md]: I only added this because I saw this message:

    Expected ${received} to be a component from react-test-renderer, or a mounted enzyme component.

[here](https://github.com/styled-components/jest-styled-components/blob/master/src/matchers/toHaveStyleRule.js#L90)

Comment (MicheleBertoli) [inline README.md]: You are absolutely right, that message should be updated.
My idea was to make it similar to the `toHaveStyleRule` matcher for RN.

File: README.md (hunk 8 above)
Comment (vamsiampolu) [inline README.md]: If you assign a style to a component from a library ( a component that you do not own), would you use toMatchSnapshot or would you try and find it, esp. if unstyled and styled elements of the same target elementt are in the subject component.

**Comment by MicheleBertoli** [comment]
🙌 Thank you very much @vamsiampolu
